coral/breakpoint/breakpoint_types.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+import io
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import intervaltree
+import numpy as np
+
+from coral.types import CnsInterval
+
+
+@dataclass
+class CNSSegData:
+    tree: intervaltree.IntervalTree
+    intervals: List[CnsInterval]
+    intervals_by_chr: Dict[str, List[List[int]]]
+    log2_cn: List[float]
+
+    @classmethod
+    def from_file(cls, file: io.TextIOWrapper) -> CNSSegData:
+        tree, log2_cns = {}, []
+        intervals = []
+        intervals_by_chr: Dict[str, List[List[Any]]] = {}
+        is_cns = file.name.endswith(".cns")
+
+        next(file)  # Skip header row
```

Comment:
There's always a header row in the inputs we expect here, which is what this previous [conditional](https://github.com/AmpliconSuite/CoRAL/blob/main/src/infer_breakpoint_graph.py#L85) was avoiding. That added an extra level of nesting to the below logic that I believe makes the code less readable, which was why I preferred this approach.

Reply:
In prior changes I had left open the option that the user provides a bed file that isn't a .cns produced by CNVkit. This is reflected in the code with the `is_cns` variable. Agreed with Matt that in either case we shouldn't assume a header. We should only skip the header if one exists - not dangerously force a skip without checking what kind of file we're reading.

I suggest modifying so that if it is a .cns file, we skip the first line (or more generically the line that starts with "chromosome"), and also we could do something like skip any line that starts with "#", since that's a general header indicator in bfx. Thoughts?

Reply:
That makes a lot of sense, thanks for clarifying. I guess another option is rather than excluding certain lines, we could only parse lines beginning with `chr*` and skip all others? Assuming that's consistent regardless of the CN generator, it might be easier than trying to enumerate all other oddities?

Reply:
GRCh37 for instance just uses chromosomes named "1, 2, 3, ...". Another common use case could be samples with oncoviral integration (HPV, EBV, etc.) which also do not start with "chr". 

I suppose you could check if fields 1 and 2 are integers and use that to help decide if the line is valid non-header or not. In my opinion, only two file formats need to be supported - .cns and .bed. For anything else I think it is the user's responsibility to deal with :)

Reply:
Reverting this to the previous check for now since no one's complained and doesn't seem like there's a perfect solution here